### ISSUE TITLE
zsh autocompletion is malformed

### ISSUE THREAD

**Issue by debris** [jwilm/alacritty#2939]
- `macOS 10.14.3`
- `alacritty 0.3.3`
- shell: oh-my-zsh: `zsh 5.3 (x86_64-apple-darwin18.0`

# Expected

when I type `cd` and then press `tab` I expect the shell suggestions to be displayed clearly

<img width="260" alt="Screenshot 2019-10-31 at 17 00 51" src="https://user-images.githubusercontent.com/2613714/67964640-e2654d00-fc00-11e9-9875-3d0861231cdb.png">

# How it is

before pressing tab:

<img width="254" alt="Screenshot 2019-10-31 at 17 08 00" src="https://user-images.githubusercontent.com/2613714/67964742-0de83780-fc01-11e9-99c6-d9230f4d4853.png">

after:

<img width="269" alt="Screenshot 2019-10-31 at 17 08 07" src="https://user-images.githubusercontent.com/2613714/67964759-16407280-fc01-11e9-92b1-6610f08e3314.png">

or: 

<img width="266" alt="Screenshot 2019-10-31 at 16 59 55" src="https://user-images.githubusercontent.com/2613714/67964768-1b9dbd00-fc01-11e9-82c7-15c3e70fcf8a.png">

**Comment by chrisduerr** [comment]
Also please confirm that the default Alacritty config does not have this issue (`alacritty --config-file /dev/null`).

**Comment by nickmass** [comment]
I am experiencing this issue as well, but it only occurs when the zsh `RPROMPT` parameter is set.

**entire .zshrc contents**
```
RPROMPT=ABC
```

**alacirtty command**
`alacritty --config-file /dev/null`

**result**
![image](https://user-images.githubusercontent.com/1085568/68152678-5c4b4e00-ff01-11e9-9670-cd1e4eff82d9.png)

The same .zshrc is handled correctly in `urxvt` and `xterm`

- `alacritty 0.3.3`
- `zsh 5.7.1`
- X11 in WSL

**Comment by debris** [comment]
> Have you confirmed this is an issue with Alacritty and not your zsh configuration? How does Terminal.app behave?

@nixpulvis It works

> Also please confirm that the default Alacritty config does not have this issue (`alacritty --config-file /dev/null`).

@chrisduerr I just checked that and I observed an interesting pattern:

- if `Alacritty` is started with or without my config file from `Finder`, `Launchpad` or another `Alacritty` instance, autocompletion is malformed
- if `Alacritty` is started with or without my config file from `Terminal.app`, autocompletion works fine

---

So comparing env variables in both terminals made me notice that env variable `LC_CTYPE` is not set in `Alacritty` and it is set to `UTF-8` in `Terminal`.

After adding `export LC_CTYPE=UTF-8` to my .zshrc autocompletion works properly 🎉 

---

It would be nice if the next version of `Alacritty` set this env variable automatically :)

**Comment by kchibisov** [comment]
fwiw you should set it in alacritty env section and not in shell.

**Comment by chrisduerr** [comment]
Did you alter the `shell` config variable @debris? Please make sure to always spawn login shells.

**Comment by kchibisov** [comment]
duplicate of #2566 (read from the beginning of the issue ).

Anyway we do have a problem with a different `LC_CTYPE` even when using a login shell.

**Comment by debris** [comment]
from #2960

> Thanks for the PR, however this is not a right way to solve this problem. We should get a proper locale from a system, otherwise it's a work around specific problem on your system.
> 
> If you'd like to work on this bug, I'd recommend you to look at our macOS specific file for setting locale https://github.com/jwilm/alacritty/blob/master/alacritty_terminal/src/locale.rs and understand why we're not getting a proper locale from a system, compared to Terminal.app. Looking at other open source terminals, which support macOS, could help here.

@kchibisov the locale we're getting is correct - `en_PL.UTF-8` and is equal to the locale returned by `defaults read -globalDomain AppleLocale`. What's failing is setting new locale in line 70.

https://github.com/jwilm/alacritty/blob/dbd8538762ef8968a493e1bf996e8693479ca783/alacritty_terminal/src/locale.rs#L69-L71

> duplicate of #2566 (read from the beginning of the issue ).
> 
> Anyway we do have a problem with a different LC_CTYPE even when using a login shell on macOS.

Yes, it is indeed a duplicate

**Comment by debris** [comment]
I also just realised that when I explicitly set env locale to `en_PL.UTF-8`, the `zsh` autocompletion is still malformed. Only setting it to `UTF-8` makes it work correctly. I should probably report this issue to `zsh` as well.

**Comment by kchibisov** [comment]
Where are you setting en_PL.UTF-8? It should be set alacritty env section, otherwise things could go wrong.

**Comment by kchibisov** [comment]
Also, we should achieve the same output of `locale`  as in `Termninal.app` (not `env`). So, if Terminal.app sets locale to `UTF-8` then we should go ahead and do the same thing, the question it should be done.

**Comment by debris** [comment]
> Where are you setting en_PL.UTF-8? It should be set in alacritty env section, otherwise things could go wrong.

I set it over there.

> Also, we should achieve the same output of locale as in Termninal.app (not env). So, if Terminal.app sets locale to UTF-8 then we should go ahead and do the same thing, the question is how it should be done.

I'll look into this later :)

**Comment by kchibisov** [comment]
Oh, would you mind to check how iterm2 behaves here? Just reading through all of these macOS locale thing, it seems like setting UTF-8 as locale is not a good idea, so probably we should do things a bit different. By not a good idea, I mean that I'm not sure how it'll affect sshing to a Linux/BSD box, since UTF-8 locale is not a thing on them iirc.

**Comment by chrisduerr** [comment]
>  By not a good idea, I mean that I'm not sure how it'll affect sshing to a Linux/BSD box, since UTF-8 locale is not a thing on them iirc.

Yeah, the `UTF-8` locale shouldn't be a thing. I don't think it is on macOS either, since it makes no sense as a locale (localization requires a country code).

**Comment by zsugabubus** [comment]
Nobody asked yet but could you please share both your left and right prompts? `zsh` have to compute how much cells are the characters take, so it's maybe (probably) just a simple counting issue that can be easily solved by fixing your prompt.

* If you use a plugin for your prompt, it's an issue with the plugin.
* If you crafted your prompt(s) manually, then you should try using [%G](http://zsh.sourceforge.net/Doc/Release/Prompt-Expansion.html#Visual-effects) and (temporary) setting `ZLE_RPROMPT_INDENT=1` (default).

**Comment by debris** [comment]
I use only one zsh plugin -> `git zsh-autosuggestions`

my `RPROMPT` is an empty string whereas my `PROMPT` is:

```
%B$PROMPTPREFIX %2~ $(custom_git_prompt)%{$M%}%B»%b%{$RESET%}
```

I did not craft it manually. it's the default prompt from zsh with the plugin that I have enabled.

It looks like the problem is already known upstream. This issue ->  https://github.com/robbyrussell/oh-my-zsh/issues/7526 seems very similar
